control-plane/agents/src/bin/core/tests/rebuild/mod.rs

```diff
@@ -184,34 +209,131 @@ async fn replica_wait_and_full_rebuild() {
+    // Check the pool on restarted node is imported back.
+    is_pool_online(&cluster, testrep, Duration::from_secs(6))
+        .await
+        .expect("Pool didn't get into online state");
 
     // Fetch replicas again.
     let replicas_now = api_client.replicas_api().get_replicas().await.unwrap();
-    // The replica picked for test must not be online now again.
-    assert!(!replicas_now.iter().any(|r| r.uri == testrep.uri));
+    // The replica picked for test must be online now again.
+    assert!(replicas_now.iter().any(|r| r.uri == testrep.uri));
+    if let Some(rep) = replicas_now.iter().find(|r| r.uri == testrep.uri) {
+        assert_eq!(rep.state, ReplicaState::Online);
+    }
 
     // Get the volume again for validations.
     let vol = volume_api.get_volume(volid.uuid()).await.unwrap();
     let volume_state = vol.state;
     let nexus = volume_state.target.unwrap();
-    // The child must not be in the nexus anymore.
+    // The child must be still in the nexus.
     let is_removed = !nexus.children.iter().any(|c| c.uri == testrep.uri);
-    assert!(is_removed);
-    // TODO: We can additionally check that the volume state eventually becomes Online,
-    // and doesn't remain degraded once the full rebuild finishes. Currently no support for
-    // getting rebuild info via nexus grpc client NexusOperations.
+    assert!(!is_removed);
+    let nexus_id = NexusId::from(volume.state.target.unwrap().uuid);
+
+    wait_nexus_online(&nexus_client, nexus_id.clone())
+        .await
+        .expect("Rebuild didnt finish in 30 secs");
+
+    let history = nexus_client
+        .get_rebuild_history(&GetRebuildRecord::new(nexus_id.clone()), None)
+        .await
+        .expect("Failed to get rebuild record");
+
+    assert_eq!(
+        nexus_id, history.uuid,
+        "Cant match nexus id in rebuild history"
+    );
+
+    assert_eq!(
+        1,
+        history.records.len(),
+        "Number of rebuild history is not equal to 1"
+    );
+
+    assert!(
+        history.records.get(0).unwrap().is_partial,
+        "Rebuild type is not Partial rebuild"
+    )
+}
+
+/// Checks if node is online, returns true if yes.
+async fn wait_nexus_online(nexus_client: &impl NexusOperations, nexus: NexusId) -> Result<(), ()> {
+    let timeout = Duration::from_secs(REBUILD_WAIT_TIME);
+    let start = std::time::Instant::now();
+    loop {
+        let nexus = nexus_client
+            .get(Filter::Nexus(nexus.clone()), None)
+            .await
+            .expect("Cant get Nexus object");
+        if let Some(nexus) = nexus.0.get(0) {
+            if nexus.status == NexusStatus::Online {
+                return Ok(());
+            }
+        }
+        if std::time::Instant::now() > (start + timeout) {
+            break;
+        }
+        tokio::time::sleep(Duration::from_millis(250)).await;
+    }
+    Err(())
+}
+
+/// Checks if nexus has rebuild record for predefined interval.
+async fn poll_rebuild_history(
```

Comment:
```suggestion
async fn wait_rebuild_history_count( count: u8,
```
And check that we have `count` records ?

Reply:
We validate number of records in test itself. After rebuild in both cases it should be 1.